pkg/scheduler/scheduler.go

```diff
@@ -170,3 +189,131 @@ func (pc *Scheduler) watchSchedulerConf(stopCh <-chan struct{}) {
 		}
 	}
 }
+
+func (pc *Scheduler) startVgpuServer() {
+	startvgpuserver := false
+	for {
+		pc.mutex.Lock()
+		plugins := pc.plugins
+		configurations := pc.configurations
+		pc.mutex.Unlock()
+		klog.V(3).Infof("Checking vGPU server start condition...")
+		ssn := framework.OpenSession(pc.cache, plugins, configurations)
+		if !pc.vgpuServerStarted {
+			for _, val := range ssn.Tiers {
+				if startvgpuserver {
+					break
+				}
+				for idx, plugins := range val.Plugins {
+					if strings.Contains(plugins.Name, predicates.PluginName) {
+						klog.Infoln("idx=", idx, "arguments=", plugins.Arguments)
+						var t framework.Arguments
+						t = plugins.Arguments
+						t.GetBool(&startvgpuserver, predicates.VGPUSharingPredicate)
+						klog.Infoln("startvgpuserver=", startvgpuserver)
+						break
+					}
+				}
+			}
+		}
+		if startvgpuserver {
+			klog.Infof("Need to startvgpuserver here")
+			pc.vgpuServerStarted = true
+			// start grpc server
+
+			lisGrpc, _ := net.Listen("tcp", vgpuutil.GrpcBind)
+			defer lisGrpc.Close()
+			s := grpc.NewServer()
+			vgpuproto.RegisterDeviceServiceServer(s, pc)
+			err := s.Serve(lisGrpc)
+			if err != nil {
+				klog.Fatal(err)
+			}
+		}
+		framework.CloseSession(ssn)
+		time.Sleep(time.Second * 5)
+	}
+}
+
+func (pc *Scheduler) Register(stream vgpuproto.DeviceService_RegisterServer) error {
+	var nodeID string
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			klog.Infof("Register Error err=", err.Error())
+		}
+		if err != nil {
+			//s.delNode(nodeID)
+			//klog.Infof("node %v leave, %v", nodeID, err)
+			_ = stream.SendAndClose(&vgpuproto.RegisterReply{})
+			return err
+		}
+		klog.V(3).Infof("device register %v", req.String())
+		nodeID = req.GetNode()
+		nodes := pc.cache.(*schedcache.SchedulerCache).Nodes
+		nodeinfo, ok := nodes[nodeID]
+		if !ok {
+			nodes[nodeID] = api.NewNodeInfo(nil)
+		}
```

Comment:
Adding an item to map here may conflicts with `SchedulerCache.addNode` in scheduler cache. How about start vgpu server and update VGPUDevices in scheduler cache.